src/utils/domUtils.ts

```diff
@@ -107,6 +103,20 @@ export module _Util {
       if (boxA.top > boxB.bottom) { return false; }
       return true;
     }
+
+    export function containedInBoundingBox(container: ClientRect, element: ClientRect) {
+      return (
+        Math.floor(container.left) <= Math.ceil(element.left) &&
+        Math.floor(container.top)  <= Math.ceil(element.top)  &&
+        Math.floor(element.right)  <= Math.ceil(container.right) &&
+        Math.floor(element.bottom) <= Math.ceil(container.bottom)
+      );
+    }
+
+    export function getBoundingSVG(elem: D3.Selection) {
+      var svg = elem[0][0].ownerSVGElement;
+      return (svg == null) ? elem : svg;
```

Comment:
Actually, there's a problem here. `ownerSVGElement` is also null if `elem` is no longer in the DOM tree, so we'll mistakenly return `elem` even though it's not an `<svg>`. This implementation addresses that:
```Typescript
export function getBoundingSVG(elem: HTMLScriptElement): HTMLScriptElement {
  var ownerSVG = elem.ownerSVGElement;
  if (ownerSVG != null) { 
    return ownerSVG;
  }

  if (elem.nodeName.toLowerCase() === "svg") { // elem itself is an SVG
    return elem;
  }

  return null; // not in the DOM
}
```
I'll also note that the implementation you had above didn't have a consistent return type...